### PR TITLE
Use new pattern language for MergeComposite

### PR DESCRIPTION
**Pull request by trevor-m**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

File: python/tvm/relay/op/contrib/tidl.py [hunk 1]
```diff
@@ -30,150 +31,112 @@ def _merge_sequential_ops(mod):
...
+        multibox_prior_out = is_op('vision.multibox.multibox_prior')(wildcard())
+        concatenate_out = is_op('concatenate')(multibox_prior_out, wildcard())
         return concatenate_out
 
     def _multibox_prior_nms_pattern():
-        x = relay.var('x')
-        y = relay.var('y')
-        multibox_prior_out = relay.op.vision.multibox.multibox_prior(x)
-        nms_out = relay.op.vision.nms.non_max_suppression(multibox_prior_out, y)
+        multibox_prior_out = relay.op.vision.multibox.multibox_prior(wildcard())
```
Comment (comaniac) [inline python/tvm/relay/op/contrib/tidl.py]: Shoudn't be `is_op`?

File: python/tvm/relay/op/contrib/tidl.py [hunk 2]
```diff
@@ -30,150 +31,112 @@ def _merge_sequential_ops(mod):
...
+        reshape_out = is_op('reshape')(transpose_out)
+        transpose_out1 = is_op('reshape')(reshape_out)
         return transpose_out1
 
     def _conv2d_relu_pattern():
-        x = relay.var('x')
-        w = relay.var('w')
-        conv2d_out = relay.op.nn.conv2d(x, w)
-        relu_out = relay.op.nn.relu(conv2d_out)
+        conv2d_out = is_op('nn.conv2d')(wildcard(), is_constant())
```
Comment (comaniac) [inline python/tvm/relay/op/contrib/tidl.py:93]: I am not sure if this is intentional so just make a note. The pattern is not equivalent to the original one. The equivalent one should be `is_op('nn.conv2d')(wildcard(), wildcard())`.

The pattern `is_op('nn.conv2d')(wildcard(), is_constant())` will only match the conv2d when the weight is bind to be a constant node.

ditto to rest cases.

**Review by comaniac — COMMENTED** [review]
Reviewed the new patterns.

File: python/tvm/relay/op/contrib/tidl.py (hunk 2 above)
Comment (trevor-m) [reply python/tvm/relay/op/contrib/tidl.py:93]: Thanks Cody, this is intentional. TIDL can only support conv2d when the weight is a constant.

File: python/tvm/relay/op/contrib/tidl.py (hunk 1 above)
Comment (trevor-m) [reply python/tvm/relay/op/contrib/tidl.py]: Oops, will fix.

File: python/tvm/relay/op/contrib/tidl.py (hunk 2 above)
Comment (comaniac) [reply python/tvm/relay/op/contrib/tidl.py:93]: Nice :) Please resolve this then.

**Review by comaniac — APPROVED** [review]

**Pull request merged**